client/post-editor/media-modal/detail/detail-fields.jsx

```diff
@@ -1,145 +1,151 @@
-	ClipboardButtonInput = require( 'components/clipboard-button-input' ),
-	FormTextarea = require( 'components/forms/form-textarea' ),
-	FormTextInput = require( 'components/forms/form-text-input' ),
-	TrackInputChanges = require( 'components/track-input-changes' ),
-	EditorMediaModalFieldset = require( '../fieldset' );
-
-module.exports = React.createClass( {
-	displayName: 'EditorMediaModalDetailFields',
-
-	propTypes: {
-		site: React.PropTypes.object,
-		item: React.PropTypes.object
-	},
-
-	getInitialState: function() {
-		return {};
-	},
-
-	componentWillMount: function() {
+import analytics from 'lib/analytics';
+import MediaUtils from 'lib/media/utils';
+import MediaActions from 'lib/media/actions';
+import ClipboardButtonInput from 'components/clipboard-button-input';
+import FormTextarea from 'components/forms/form-textarea';
+import FormTextInput from 'components/forms/form-text-input';
+import TrackInputChanges from 'components/track-input-changes';
+import EditorMediaModalFieldset from '../fieldset';
+
+class EditorMediaModalDetailFields extends Component {
+	static propTypes = {
+		site: PropTypes.object,
+		item: PropTypes.object
+	};
+
+	state = {};
+
+	componentWillMount() {
 		this.persistChange = debounce( this.persistChange, 1000 );
-	},
-
-	componentWillReceiveProps: function( nextProps ) {
-		if ( this.props.item && nextProps.item && nextProps.item.ID !== this.props.item.ID ) {
-			this.setState( {
-				modifiedItem: null
-			} );
+	}
 
+	componentWillReceiveProps( nextProps ) {
+		if (
+			this.props.item &&
+			nextProps.item &&
+			nextProps.item.ID !== this.props.item.ID
+		) {
+			this.setState( { modifiedItem: null } );
 			this.persistChange.cancel();
 		}
-	},
-
-	bumpStat: function( stat ) {
-		switch ( stat ) {
-			case 'alt':
-				analytics.ga.recordEvent( 'Media', 'Changed Image Alt' );
-				break;
+	}
 
-			case 'caption':
-				analytics.ga.recordEvent( 'Media', 'Changed Item Caption' );
-				break;
+	bumpAltStat = () => {
+		analytics.ga.recordEvent( 'Media', 'Changed Image Alt' );
+		analytics.mc.bumpStat( 'calypso_media_edit_details', 'alt' );
+	};
 
-			case 'description':
-				analytics.ga.recordEvent( 'Media', 'Changed Item Description' );
-				break;
-		}
+	bumpCaptionStat = () => {
+		analytics.ga.recordEvent( 'Media', 'Changed Item Caption' );
+		analytics.mc.bumpStat( 'calypso_media_edit_details', 'caption' );
+	};
 
-		analytics.mc.bumpStat( 'calypso_media_edit_details', stat );
-	},
+	bumpDescriptionStat = () => {
+		analytics.ga.recordEvent( 'Media', 'Changed Item Description' );
+		analytics.mc.bumpStat( 'calypso_media_edit_details', 'description' );
+	};
 
-	isMimePrefix: function( prefix ) {
+	isMimePrefix( prefix ) {
 		return MediaUtils.getMimePrefix( this.props.item ) === prefix;
-	},
+	}
 
-	persistChange: function() {
+	persistChange() {
 		if ( ! this.props.site || ! this.state.modifiedItem ) {
 			return;
 		}
 
-		debug( 'Update media to %o', this.state.modifiedItem );
```

Comment:
There is only one line bug in this component. It doesn't add too much information regarding the debugging process, especially keeping in mind that we can use the react tab.
Finally, I think that little by little, we are not using debug in our base code. Keep in mind this code is quite old according to the calypso history. Especially, this line is from the beginning.

But ... I could restore it if we consider that make sense keep it there. Not a big deal at all.